include/hip/hcc_detail/hip_runtime_api.h

```diff
@@ -1442,14 +1506,50 @@ hipError_t hipGetSymbolSize(size_t* size, const void* symbolName);
  * hipMemcpyFromArrayAsync, hipMemcpy2DFromArrayAsync, hipMemcpyToSymbolAsync,
  * hipMemcpyFromSymbolAsync
  */
-hipError_t hipMemcpyToSymbolAsync(const void* symbolName, const void* src, size_t sizeBytes,
-                                  size_t offset, hipMemcpyKind kind, hipStream_t stream __dparm(0));
+inline
+__attribute__((visibility("hidden")))
+hipError_t hipMemcpyToSymbolAsync(const void* symbolName, const void* src,
```

Comment:
Implementing the hipMemcpyToSymbol* APIs as inline functions in a header won't work with the dlsym() use case that some users are asking for, can this be moved into a .cpp file?